desc/coils.py

```diff
@@ -798,13 +798,27 @@ class CoilSet(OptimizableCollection, _Coil, MutableSequence):
         assuming 'virtual' coils from the other half field period. Default = False.
     name : str
         Name of this CoilSet.
-
+    check_intersection: bool
+        Whether or not to check the coils in the coilset for intersections.
+    check_grid: Grid, optional
```

Comment:
Do we need this? I feel like we can use the default grid for the automatic checking. For comparison, eq.is_nested allows custom grids, but the automatic check that happens as part of init always uses the default